Generate systems from get tagging resources call

earmenda opened the pull request:
Closes #575

### Code Changes

* [x] Added `get_resources` call from `resourcegroupstaggingapi` api for aws connector
* [x] Added `create_resource_tagging_systems` function that takes list of arns and returns systems
* [x] Added arn system generation for dynamodb tables and s3 buckets

### Steps to Confirm

* [x] Manual testing using `generate` command
* [x] Integration testing

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation Updated:
  - [x] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [x] documentation issue created (tag docs-team to complete issue separately)
* [x] Issue Requirements are Met
* [x] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`

### Description Of Changes

* Decided to simplify this implementation by just adding some resources we did not have using the response from the `resourcegroupstaggingapi` api. We could expand this to return any systems which can be generated from an arn that this api returns.

ThomasLaPiana commented on src/fidesctl/ctl/connectors/aws.py:
```diff
@@ -82,6 +82,20 @@ def describe_rds_instances(client: Any) -> Dict[str, List[Dict]]:  # type: ignor
 
 
+@handle_common_aws_errors
+def get_tagging_resources(client: Any) -> List[str]:  # type: ignore
```
is there any way to put an actual type here?

earmenda replied:
The boto3 library does not return a typed client when calling `boto3.client` unfortunately. 

https://github.com/boto/botocore/issues/2297

ThomasLaPiana commented on src/fidesctl/ctl/connectors/aws.py:
```diff
@@ -162,3 +176,80 @@ def create_rds_systems(
+) -> Optional[System]:
+    """
+    Given an AWS arn for a s3 resource, returns a System representation
+    if System is desired.
```
what does `if System is desired` mean in this context?

earmenda replied:
An aws service could have different resource types. So for instance for dynamodb we only create systems for tables but not necessarily for auto scaling policies.

earmenda replied:
Updating the comment to make this more clear!

ThomasLaPiana replied:
ohhh got it! thanks for the clarification

ThomasLaPiana commented on src/fidesctl/ctl/connectors/aws.py:
```diff
@@ -82,6 +82,20 @@ def describe_rds_instances(client: Any) -> Dict[str, List[Dict]]:  # type: ignor
+@handle_common_aws_errors
+def get_tagging_resources(client: Any) -> List[str]:  # type: ignore
+    """
+    Returns a list of resource arns given a resourcegroupstaggingapi boto3 client.
```
```suggestion
    Returns a list of resource arns given a 'resourcegroupstaggingapi' boto3 client.
```

earmenda replied:
done

ThomasLaPiana reviewed (changes requested):
Code-wise super clean changes here!
Looks like it still needs a CHANGELOG entry and there might be some missing documentation updates? To the generation docs specifically

earmenda commented:
> Code-wise super clean changes here! Looks like it still needs a CHANGELOG entry and there might be some missing documentation updates? To the generation docs specifically

Thanks! Updated relevant docs and changelog. Addressed all comments, should be ready for merge.
